### ISSUE TITLE
The DPR results

### ISSUE THREAD

**Issue by xionghao132** [Hannibal046/nanoDPR#1]
Thank you for sharing your code and results, I find  the new DPR can achieve 87.3 at top-100，while I get 85.9 the same as your results. Limited by my device, I change the batch_size=6, val_batch_size=6, and run on 6 gpus. I read the DPR paper, the number of negative documents may have an influence on the result, so do you know the reason?

**Comment by Hannibal046** [comment]
Hi @xionghao132,
Thanks for your interest and feedback on nanoDPR. 
The reason behind the performance gap of top-100 accuracy, I believe, is the training data. As stated in the original DPR repo, the 87.3 top-100 accuracy is obtained by using the combination of original training data and adversarially mined negative samples. In the contrastive learning process, the quality of negative samples matters. 

Also, I think the train-batch-size is not the key to the performance gap here.

**Comment by xionghao132** [comment]
Thank you for your reply, but I have already combined the original data and the new mind data. So you get the results just on original data?

**Comment by Hannibal046** [comment]
yes, i got the results only on the original training data without mined ones as you can see in the config file:
https://github.com/Hannibal046/nanoDPR/blob/c97f996efdff457ba03608a7d1b9b1dee25cabaa/config/train_dpr_nq.yaml#L2

Given this, I think the problem was on the `per_device_train_batch_size`. Because the current implementation would gather all samples from all GPU cards. So in my case, the negative samples for one query is much larger than yours.

**Comment by xionghao132** [comment]
OK, thank you for your patience in answering.
